Task2/Task2_1/Task2_1_1/CustomString.cs

```diff
@@ -0,0 +1,166 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Task2_1_1
+{
+    class CustomString
+    {
+        private char[] _charString;
+
+        public int Length { get => _charString.Length; }
+
+        public char this[int index]
+        {
+            get => _charString[index];
+            set => _charString[index] = value;
+        }
+
+        public CustomString()
+        {
+            _charString = new char[0];
+        }
+
+
+        public CustomString(params char[] symbols)
+        {
+            _charString = new char[symbols.Length];
+
+            for (int i = 0; i < _charString.Length; i++)
+            {
+                _charString[i] = symbols[i];
+            }
+        }
+
+        public override bool Equals(object obj)
+        {
+            if (Object.ReferenceEquals(this, obj)) return true;
+            if (this.GetHashCode() == obj.GetHashCode()) return true;
+            if (!(obj is CustomString) || obj == null) return false;
+            if (this.Length != ((CustomString)obj).Length) return false;
+
+            for (int i = 0; i < this.Length; i++)
+            {
+                if (this[i] != ((CustomString)obj)[i]) return false;
+            }
+
+            return true;
+        }
```

Comment:
```suggestion
        public override bool Equals(object obj)
        {
            if (obj == null || !(obj is CustomString) || this.Length != ((CustomString)obj).Length)
            {
                return false;
            }
             
             var argument = obj as CustomString;

            if (Object.ReferenceEquals(this, argument) || this.GetHashCode() == argument.GetHashCode())
            {
                return true;
            }

            for (int i = 0; i < this.Length; i++)
            {
                if (this[i] != ((CustomString)obj)[i]) return false;
            }

            return true;
        }
```

Так будет правильнее с точки зрения читаемости, плюс в случае с "||" проверка не пойдет дальше, если какое-то из условий вернет true - т.е. если аргумент нуловый, то он сразу вернет истину и дальше проверять не будет, аналогично и с положительным - если ссылки одинаковые, то проверять по хешам он не будет. Еще у меня сомнения, что в твоем случае сравнение хешей будет работать, т.к. у тебя будут разные имплементации GetHashCode (для твоего типа переопределен).